test/e2e/credit.test.ts

```diff
@@ -0,0 +1,54 @@
+import { expect } from "chai"
+import { Connection, connect } from "../../src"
+import { Rabbit } from "../support/rabbit"
+import { eventually } from "../support/util"
+import { Offset } from "../../src/requests/subscribe_request"
+import { Message } from "../../src/producer"
+
+describe("credit management", () => {
+  const rabbit = new Rabbit()
+  const streamName = "credit-test-stream"
+  let connection: Connection
+
+  beforeEach(async () => {
+    connection = await connect({
+      hostname: "localhost",
+      port: 5552,
+      username: "rabbit",
+      password: "rabbit",
+      vhost: "/",
+      frameMax: 0, // not used
+      heartbeat: 0, // not used
+      listeners: {
+        metadata_update: (_data) => console.info("Subscribe server error"),
+      },
+    })
+    await rabbit.createStream(streamName)
+  })
+
+  afterEach(async () => {
+    await connection.close()
+    await rabbit.deleteStream(streamName)
+  })
+
+  // This test can only run locally, the HTTP API gives different results in GitHub CI (https://coders51.slack.com/archives/C03E263HH38/p1681374600592829)
+  it.skip(`the number of credit remain stable after have consumed some messages`, async () => {
```

Comment:
This test fail because (I think) there is a bug in the consumer see the pr #86. But it's fine to skip it for now. Can you create a card to fix this test in the feature?